readingDataPointsFilters:
  - RandomSamplingDataPointsFilter:
      prob: 0.5

referenceDataPointsFilters:
  - SamplingSurfaceNormalDataPointsFilter:
      knn: 10

matcher:
  KDTreeMatcher:
    knn: 1
    epsilon: 0 

outlierFilters:
  - TrimmedDistOutlierFilter:
      ratio: 0.75

errorMinimizer:
  PointToPlaneErrorMinimizer

transformationCheckers:
  - CounterTransformationChecker:
      maxIterationCount: 1000
  - DifferentialTransformationChecker:
      minDiffRotErr: 0.0005
      minDiffTransErr: 0.01
      smoothLength: 5   

inspector:
 NullInspector

# inspector:
#  VTKFileInspector:
#      baseFileName: pointmatcher-run1
#      dumpPerfOnExit: 0
#      dumpStats: 0
#      dumpIterationInfo: 1 
#      dumpDataLinks: 0 
#      dumpReading: 0 
#      dumpReference: 0 


logger:
  NullLogger
#  FileLogger

